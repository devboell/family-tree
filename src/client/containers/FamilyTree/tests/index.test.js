import React from 'react'
import { FamilyTree } from '../index'

describe('FamilyTree component', () => {
  it('loading TRUE familyTree EMPTY, renders without crashing', () => {
    const props = {
      loading: true,
      familyTree: {},
    }
    expect(shallow(<FamilyTree {...props} />)).toMatchSnapshot()
  })

  it('loading FALSE familyTree EMPTY, renders without crashing', () => {
    const props = {
      loading: false,
      familyTree: {},
    }
    expect(shallow(<FamilyTree {...props} />)).toMatchSnapshot()
  })

  it('loading FALSE familyTree NOT EMPTY, renders without crashing', () => {
    const props = {
      loading: false,
      familyTree: {
        id: '1',
        name: 'John',
        partners: [],
      },
    }
    expect(shallow(<FamilyTree {...props} />)).toMatchSnapshot()
  })
})
