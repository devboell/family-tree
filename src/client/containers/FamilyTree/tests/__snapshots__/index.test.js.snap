// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FamilyTree component loading FALSE familyTree EMPTY, renders without crashing 1`] = `<div />`;

exports[`FamilyTree component loading FALSE familyTree NOT EMPTY, renders without crashing 1`] = `
<SVGTree
  dimensions={
    Object {
      "boxHeight": 60,
      "boxWidth": 80,
      "marginX": 30,
      "marginY": 50,
    }
  }
  treeData={
    Object {
      "level": 0,
      "partners": Array [],
      "totalHeight": 110,
      "totalWidth": 110,
      "value": "John",
    }
  }
/>
`;

exports[`FamilyTree component loading TRUE familyTree EMPTY, renders without crashing 1`] = `
<div>
  Loading ...
</div>
`;
